e2e_testing/README.md

```diff
@@ -1 +1,95 @@
+
+We are using [Playwright](https://playwright.dev/) as the testing framework. This provides browser instances augmented with Playwright's language API for selecting DOM elements, emulating user interactions and making assertions.
+
+## Testing a User's Perspective
+
+A primary benefit of E2E Testing (also Acceptance Testing in this context) is that the tests have zero or very limited visibility of the internals of the system under test. This yields a focus on the user facing feature set irrespective of the code, facilitating refactor without test rewrite and producing a test suite styled around the product requirements and a test output that describes the product's specification.
+
+The tests then should interpret the rendered UI as would a user or an assistive technology. Historically source code would be annotated with HTML attributes for the tests to select on, or tests would make use of DOM XPaths or CSS selector paths. These approaches should be avoided as they incur a reliance on the implementation detail, can be brittle and do not reflect a user's view of the system. Playwright's best practices include a recommendation to [test user-visible behavior](https://playwright.dev/docs/best-practices#test-user-visible-behavior) and an element [Locator](https://playwright.dev/docs/locators) preference to select on accessibility attributes and textual content. Where practical, this approach should be followed.
+
+## Development
+
+Playwright provides a [UI workbench](https://playwright.dev/docs/test-ui-mode) for running and debugging tests during development. This can be started from the `e2e_testing` directory with:
+
+```bash
+yarn start
+```
+
+For headless testing from the command line and on CI, we can run the tests with:
+
+```bash
+yarn test
+```
+
+To target specific environments, set `BASE_URL` on the environment, e.g:
+
+```bash
+BASE_URL=https://mitopen.odl.mit.edu/ yarn test
+```
+
+NPM scripts are provided for our RC and Production environments:
+
+```bash
+yarn test:rc
+yarn test:production
+```
+
+A Docker Compose configuration at [../docker-compose-e2e-tests.yml](../docker-compose-e2e-tests.yml) provides the minimal run configuration for starting services needed for testing, the Postgres database, the backend web service and our nginx reverse proxy.
+
+Services can be started with:
+
+```bash
+docker compose -f docker-compose-e2e-tests.yml up web db nginx
+```
+
+The tests themselves are also containerized so the following command will start all services, run the tests and shut down services when the tests have completed:
+
+```bash
+docker compose -f docker-compose-e2e-tests.yml up --exit-code-from e2e-tests
+```
+
+Note that to be loaded from within the e2e-tests container, the front end must first be built for production (in development mode Webpack builds the static assets URLs to `od.odl.local`, not available on the Docker network).
+
+```bash
+NODE_ENV=production yarn build
+```
```

Comment:
Maybe add this for docker:

"
Or build with docker (using `NODE_ENV=production` in your .env file):
```bash
docker compose run --rm watch yarn run build 
```
"

Reply:
I've added a `build-frontend` config, similar to the `watch` container, but exits after build instead of watching for changes. I haven't quite got it to work on CI with a single build-and-run command ([numerous attempts today](https://github.com/mitodl/mit-open/actions?query=actor%3Ajonkafton)), though it exists for anyone having issues with local `yarn` commands.

```bash
docker compose -f docker-compose-e2e-tests.yml run build-frontend
```